src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -654,6 +660,20 @@ private void genConstructor(ClassBuilder clb) {
                               .return_());
         }
 
+        private void genConstants(ClassBuilder clb) {
+            var cinitSnippets = dedupSetBuilder.buildConstants(clb);
+            if (!cinitSnippets.isEmpty()) {
+                clb.withMethodBody(
+                        CLASS_INIT_NAME,
+                        MTD_void,
+                        ACC_PUBLIC | ACC_STATIC,
```

Comment:
```suggestion
                        ACC_STATIC,
```
By convention, we simply set the static flag without anything else set.